dbt-metricflow/dbt_metricflow/cli/cli_configuration.py

```diff
@@ -17,25 +17,37 @@
 logger = logging.getLogger(__name__)
 
 
-class CLIContext:
+class CLIConfiguration:
     """Context for MetricFlow CLI."""
 
-    def __init__(self) -> None:
-        """Initialize the CLI context for executing commands.
-
-        The dbt_artifacts construct must be loaded in order for logging configuration to work correctly.
-        """
+    def __init__(self) -> None:  # noqa: D107
         self.verbose = False
-        self._dbt_project_metadata: dbtProjectMetadata = dbtProjectMetadata.load_from_project_path(pathlib.Path.cwd())
+        self._dbt_project_metadata: Optional[dbtProjectMetadata] = None
         self._dbt_artifacts: Optional[dbtArtifacts] = None
         self._mf: Optional[MetricFlowEngine] = None
         self._sql_client: Optional[SqlClient] = None
         self._semantic_manifest: Optional[SemanticManifest] = None
         self._semantic_manifest_lookup: Optional[SemanticManifestLookup] = None
+
+    def setup(self) -> None:
+        """Setup this configuration for executing commands.
+
+        The dbt_artifacts construct must be loaded in order for logging configuration to work correctly.
+        """
+        dbt_project_path = pathlib.Path.cwd()
+        self._dbt_project_metadata = dbtProjectMetadata.load_from_project_path(dbt_project_path)
+
         # self.log_file_path invokes the dbtRunner. If this is done after the configure_logging call all of the
         # dbt CLI logging configuration could be overridden, resulting in lots of things printing to console
         self._configure_logging(log_file_path=self.log_file_path)
 
+    def _get_dbt_project_metadata(self) -> dbtProjectMetadata:
```

Comment:
do we need this when we already have the `@property.dbt_project_metadata`? Doesn't seem like it's used